### PR TITLE
[vjap-25] [vjap-26] 

### PR DESCRIPTION
**Pull request by egrozdani**
fixed sonar vulnerabilities, moved common security configuration to a separate module, fixed spodbugs path traversal owasp finding

### CODE REVIEW

File: security/src/main/java/ch/valtech/kubernetes/microservice/cluster/security/config/KafkaTopicConfiguration.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package ch.valtech.kubernetes.microservice.cluster.filestorage.config;
+package ch.valtech.kubernetes.microservice.cluster.security.config;
```
Comment (lrohner) [inline security/src/main/java/ch/valtech/kubernetes/microservice/cluster/security/config/KafkaTopicConfiguration.java:1]: I don't think we should move Kafka Config, it's not security related.

File: functions/src/main/java/ch/valtech/kubernetes/microservice/cluster/function/Config.java [hunk 2]
```diff
@@ -32,7 +32,7 @@ public static void main(String[] args) {
       try {
         String reversed = StringUtils.reverse(value);
         if (context != null) {
-          context.getLogger().info(String.format("Function: %s is reversing %s", context.getFunctionName(), value));
+          context.getLogger().info(() -> "Function: " + context.getFunctionName() + " is reversing " + value);
```
Comment (lrohner) [inline functions/src/main/java/ch/valtech/kubernetes/microservice/cluster/function/Config.java:35]: Why this function?

File: security/build.gradle [hunk 3]
```diff
@@ -0,0 +1,57 @@
+plugins {
+    id 'org.springframework.boot' version '2.4.5'
```
Comment (lrohner) [inline security/build.gradle:2]: You can remove the boot plugin as we don't build a bootJar

File: security/build.gradle [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+sourceCompatibility = 15
+targetCompatibility = 15
+
+configurations {
+    compileOnly {
+        extendsFrom annotationProcessor
+    }
+}
+
+bootJar {
```
Comment (lrohner) [inline security/build.gradle:30]: And remove this after removing boot plugin

File: security/build.gradle [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    }
+}
+
+dependencies {
+    implementation 'org.springframework.boot:spring-boot-starter-security'
+    implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'org.springframework.boot:spring-boot-starter-oauth2-resource-server'
+    implementation 'commons-io:commons-io:2.8.0'
+    implementation 'org.apache.commons:commons-lang3:3.12.0'
+    implementation 'net.logstash.logback:logstash-logback-encoder:6.6'
```
Comment (lrohner) [inline security/build.gradle:50]: logback not need as just a library project

File: security/build.gradle [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    implementation 'org.springframework.kafka:spring-kafka:2.7.0'
```
Comment (lrohner) [inline security/build.gradle:51]: Same kafka, should not go to security

**Review by lrohner — CHANGES_REQUESTED** [review]

File: functions/src/main/java/ch/valtech/kubernetes/microservice/cluster/function/Config.java (hunk 2 above)
Comment (egrozdani) [reply functions/src/main/java/ch/valtech/kubernetes/microservice/cluster/function/Config.java:35]: It is a code smell by Sonar https://sonar.infocentric.ch/project/issues?id=microservice-kubernetes-cluster&open=AXjLGJbigiVO85PxN4BD&resolved=false&sinceLeakPeriod=true and one of the compliant solutions is to use a supplier which is evaluated lazily

**Pull request merged**